File: symex/f_byhand.go
package main

func f(x int32) int32 {
	arr := [4]int32{1, 2, 0, 4}

	if x >= 0 && x < 4 {
		return 10 / arr[x]
	}

	return -1
}


//go:wasm-module owi
//export i32_symbol
func i32_symbol() int32

//go:export check
func check() {
	var x int32 = i32_symbol()
	var _ int32 = f(x)
}

// main is required by the tinygo compiler, even if it isn't used.
func main() {}

File: doc/src/symex/f_byhand.zig
fn f(x: usize) i32 {

  const arr = [_]i32{ 1, 2, 0, 4 };

  if (x < arr.len) {
    return @divTrunc(10, arr[x]);
  }

  return -1;
}

extern "owi" fn i32_symbol() usize;

export fn check() void {
  const x: usize = i32_symbol();
  _ = &f(x);
}
